setup.cfg

```diff
@@ -0,0 +1,71 @@
+[metadata]
+description = pathlib api extended to use fsspec backends
```

Comment:
Why is this metadata moving here from the pyproject.toml?

Reply:
The main reason is: "I'm just following the template", which removes usage of `flit` and uses setuptools instead.

Then there's of course the question: _Why don't we configure setuptools via pyproject.toml_?
1. while setuptools supports using `[project]` metadata, some of the other configuration we need is currently still considered `beta` in `pyproject.toml`.
   See: [setuptools.config.pyprojecttoml](https://github.com/pypa/setuptools/blob/09e501feb13e7dd42cc62c7f02e32c4073d61a22/setuptools/config/pyprojecttoml.py#L477-L479) and [setuptools docs](https://setuptools.pypa.io/en/latest/userguide/pyproject_config.html#setuptools-specific-configuration)
2. this way all the package specific metadata just lives in setup.cfg and the tool configuration (not flake8 due to missing support) in pyproject.toml

EDIT:
I wrote this, and realized that I was still shipping a `.flake8` config. So: I merged the flake8 configuration into the `.flake8` file.

Now, we have all package relevant configuration in `setup.cfg` and all tool relevant configuration in `pyproject.toml` (and `.flake8`, until at some point it might support toml).